### PR TITLE
Use environment config

### PR DESCRIPTION
**Pull request by AdamHawtin**
# Motivation and Context
Sets up a way of reading config from the environment, pattern taken from https://github.com/ONSdigital/dp-import-api/tree/develop/config

# What has changed
* Add config and tests
* Use build tag to ignore integration test in unit test run

# How to test?
Run the tests, build the docker image and run in with `make up` then try posting a test message in and see it process.

# Links
https://trello.com/c/rA7COLKd/871-pubsub-adapter-read-and-check-configuration-from-environment-5

### CODE REVIEW

**Comment by AdamHawtin** [comment]
[Pubsub adapter - Read and check configuration from environment (5?)](https://trello.com/c/rA7COLKd/871-pubsub-adapter-read-and-check-configuration-from-environment-5)

File: config/config.go [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+	"fmt"
+	"github.com/kelseyhightower/envconfig"
+)
+
+// Configuration structure which hold information for configuring the import API
+type Configuration struct {
+	RabbitHost             string `envconfig:"RABBIT_HOST" required:"true"`
+	RabbitPort             string `envconfig:"RABBIT_PORT" required:"true"`
+	RabbitUsername         string `envconfig:"RABBIT_USERNAME" required:"true"`
+	RabbitPassword         string `envconfig:"RABBIT_PASSOWRD"  required:"true"  json:"-"`
```
Comment (hbrace) [inline config/config.go]: Misspelling on RABBIT_PASSOWRD

File: pubsub-adapter-compose.yml [hunk 2]
```diff
@@ -6,4 +6,9 @@ services:
     environment:
       - PUBSUB_EMULATOR_HOST=pubsub-emulator-pubsub-adapter-dev:8539
       - RABBIT_CONNECTION=amqp://guest:guest@rabbitmq-pubsub-adapter-dev:5672/
+      - RABBIT_USERNAME=guest
+      - RABBIT_PASSWORD=guest
+      - RABBIT_HOST=rabbitmq-pubsub-adapter-dev
+      - RABBIT_PORT=5672
+      - RABBIT_VHOST=/
```
Comment (ryangrundy7) [inline pubsub-adapter-compose.yml]: Missing the eQ project and subcription?

File: development.env [hunk 3]
```diff
@@ -0,0 +1,8 @@
+RABBIT_HOST=localhost
```
Comment (ryangrundy7) [inline development.env]: Is this loaded in anywhere or just for us to know what the defaults are?

Comment (AdamHawtin) [reply development.env]: It's not used automatically. Could put it in the README instead of it's own file if you think that's clearer?

Comment (ryangrundy7) [reply development.env]: Yeah that looks better 👍

**Review by hbrace — APPROVED** [review]

**Review by ryangrundy7 — APPROVED** [review]
LGTM 👍

File: development.env (hunk 3 above)
Comment (AdamHawtin) [reply development.env]: I woud like to have it default to dev values locally but I could see an immediate way of doing this nicely and I thought the priority should be having it require proper env config in prod settings, might revisit this later to see if we can make the dev config behaviour a bit nicer.

File: config/config.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
+package config
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/kelseyhightower/envconfig"
+)
+
+// Configuration structure which hold information for configuring the import API
```
Comment (davidmort) [inline config/config.go]: assuming this was copied from DP's import API, prob should change this comment

File: pubsub-adapter-compose.yml [hunk 5]
```diff
@@ -6,4 +6,11 @@ services:
     environment:
```
Comment (davidmort) [inline pubsub-adapter-compose.yml]: do we need a separate compose file here, or can we have just the one compose file and use commands like ```docker-compose -f <compose.yml> up -d <individual-service>``` to bring up individual services

**Review by davidmort — CHANGES_REQUESTED** [review]

**Review by davidmort — APPROVED** [review]

**Pull request merged**